state/unit.go

```diff
@@ -1780,7 +1780,7 @@ func unitMachineStorageParams(u *Unit) (*machineStorageParams, error) {
 			return nil, errors.Annotatef(err, "getting storage instance")
 		}
 		machineParams, err := machineStorageParamsForStorageInstance(
-			u.st, chMeta, u.UnitTag(), u.Series(), allCons, storage,
+			u.st, chMeta, u.UnitTag(), allCons, storage,
```

Comment:
This is referring to a path for another machine; thus we need to use series/OS to determine the correct set of paths.